book/specifications/kimchi/template.md

```diff
@@ -158,6 +183,44 @@ TODO: linearization
 
 ### Lookup
 
+Lookups in kimchi allows you to check if a single value, or a series of values, are part of a table.
+The first case is useful to check for checking if a value belongs to a range (from 0 to 1,000, for example), whereas the second case is useful to check truth tables (for example, checking that three values can be found in the rows of an XOR table) or write and read from a memory vector (where one column is an index, and the other is the value stored at that index).
+
+```admonish
+Similarly to the generic gate, each values taking part in a lookup can be scaled with a fixed field element.
+```
+
+The lookup functionality is an opt-in feature of kimchi that can be used by custom gates.
+From the user's perspective, not using any gates that make use of lookups means that the  feature will be disabled and there will be no overhead to the protocol.
+
+```admonish
+For now, the Chacha gates are the only gates making use of lookups.
+```
+
+Refer to the [lookup RFC](../rfcs/3-lookup.md) for an overview of the lookup feature.
+
+In this section, we describe the tables kimchi supports, as well as the different lookup selectors (and their associated queries)
+
+#### The Lookup Tables
+
+Kimchi currently supports a single lookup table: 
+
+{sections.tables}
+
+**XOR**. {sections.xor}
+
+#### The Lookup Selectors
+
+**ChaChaSelector**. Performs 4 queries to the XOR lookup table.
```

Comment:
I was thinking of an interesting design in a future where we might have lots of different queries (cc @mrmr1993):

* A group of queries can be defined as expressions like "**A**: performs 4 XOR lookups on the following registers"
* a gate advertise its use of lookup as "I'm using the group of queries **A** on current row and/or next row"
* at setup time, we figure out how many queries are used, and we optimize so that:
  - if a gate is the only one using a group of queries, and its only used on the current row, then make that gate the selector
  - allocate the right number of lookup selectors for the rest (which is what we do know IIUC)

in that sense, we should perhaps specify this section as "queries patterns" or "queries groups" or another more descriptive name